src/control/server/ctl_storage.go

```diff
@@ -246,13 +306,84 @@ func (c *StorageControlService) ScmPrepare(req storage.ScmPrepareRequest) (*stor
 	return c.storage.Scm.Prepare(req)
 }
 
-// NvmeScan scans locally attached SSDs and bypasses cache.
-func (c *StorageControlService) NvmeScan(req storage.BdevScanRequest) (*storage.BdevScanResponse, error) {
-	req.BypassCache = true
-	return c.storage.ScanBdevs(req)
-}
-
 // ScmScan scans locally attached modules, namespaces and state of DCPM config.
 func (c *StorageControlService) ScmScan(req storage.ScmScanRequest) (*storage.ScmScanResponse, error) {
 	return c.storage.Scm.Scan(req)
 }
+
+// NvmePrepare preps locally attached SSDs and returns error.
+//
+// Suitable for commands invoked directly on server, not over gRPC.
+func (c *StorageControlService) NvmePrepare(req storage.BdevPrepareRequest) (*storage.BdevPrepareResponse, error) {
+	return c.storage.PrepareBdevs(req)
+}
+
+// mapCtrlrs maps each controller to it's PCI address.
+func mapCtrlrs(ctrlrs storage.NvmeControllers) (map[string]*storage.NvmeController, error) {
+	ctrlrMap := make(map[string]*storage.NvmeController)
+
+	for _, ctrlr := range ctrlrs {
+		if _, exists := ctrlrMap[ctrlr.PciAddr]; exists {
+			return nil, errors.Errorf("duplicate entries for controller %s",
+				ctrlr.PciAddr)
+		}
+
+		ctrlrMap[ctrlr.PciAddr] = ctrlr
+	}
+
+	return ctrlrMap, nil
+}
+
+// scanAssignedBdevs retrieves up-to-date NVMe controller info including
+// health statistics and stored server meta-data. If I/O Engines are running
+// then query is issued over dRPC as go-spdk bindings cannot be used to access
+// controller claimed by another process. Only update info for controllers
+// assigned to I/O Engineslled.
```

Comment:
nit - typo

Reply:
thanks fix in subsequent PR

Reply:
fix in https://github.com/daos-stack/daos/pull/6366